woocommerce/woocommerce
# fix: refactored core profiler plugins page and added error screen

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR contains several goals:
1. Refactor the plugins installation page so that it doesn't get overly complex to add the required changes
2. Fix some minor UI issues (e.g the error banner being wider than the cards)
3. Add a pre-plugins page `install_plugin` permissions check
4. Add a new child state for the plugins state, that represents the case when the permissions check fails
5. Add tracks for this new state
6. Unit tests for all the above
7. Added storybook stories for above

Closes https://github.com/woocommerce/woocommerce/issues/51763



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Spin up a site with this PR (e.g using jurassic ninja)
2. To test the no permissions page, add a new user before going through the core profiler, assigning it the role of "Shop manager"
3. Go through the full core profiler with the shop manager in an incognito window, and observe the new no permissions page
    - if you click on continue it should let you exit the core profiler after the loading screen
5. Go through the full core profiler with the full admin user and assert that plugins installation works as previously
    - clicking on anywhere within the card should toggle the checkboxes
    - clicking on already installed plugins should not have any effect
    - the learn more link still works
    - if no plugins are selected it should not install any plugins
    - selected plugins should still work as before

No permissions plugins page:
![image](https://github.com/user-attachments/assets/846edc72-6fdf-445d-a503-4d937462cd11)

Regular plugins page:
![image](https://github.com/user-attachments/assets/6377d532-2c63-4e66-b3da-eb067392da14)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
